lib/ui/widget/custom_page.dart

```diff
@@ -41,17 +44,28 @@ class CustomPage extends Page {
 ///
 /// Uses a [FadeUpwardsPageTransitionsBuilder] on Android.
 class _CupertinoPageRoute<T> extends PageRoute<T> {
-  _CupertinoPageRoute({super.settings, required this.pageBuilder})
-    : matchingBuilder = PlatformUtils.isAndroid
-          ? const FadeUpwardsPageTransitionsBuilder()
-          : const CustomCupertinoPageTransitionsBuilder();
+  _CupertinoPageRoute({
+    super.settings,
+    required this.pageBuilder,
+    required bool instantTransition,
+  }) : _transitionDuration = instantTransition
+           ? Duration.zero
+           : const Duration(milliseconds: 400),
+       matchingBuilder = PlatformUtils.isAndroid
+           ? const FadeUpwardsPageTransitionsBuilder()
+           : const CustomCupertinoPageTransitionsBuilder();
 
   /// [PageTransitionsBuilder] transition animation.
   final PageTransitionsBuilder matchingBuilder;
 
   /// Builder building the [Page] itself.
   final RoutePageBuilder pageBuilder;
 
+  /// Custom transition duration.
+  ///
+  /// Used to build instant transitioning on Desktop.
+  final Duration _transitionDuration;
```

Comment:
You can just do

```dart
@override
final Duration transitionDuration;
```

Instead of defining a private `_transitionDuration` and returning it in getter. Setter of `transitionDuration` won't be available anyway, since the variable is `final`.